pkg/data/generic.go

```diff
@@ -39,35 +38,27 @@ const (
 )
 
 // New returns a new GenericSync that cna be started.
-func New(kubeconfig *rest.Config, opa opa_client.Data, ns types.ResourceType) *GenericSync {
+func New(client dynamic.Interface, opa opa_client.Data, ns types.ResourceType) *GenericSync {
```

Comment:
I suppose we don't need to worry about compatibility here...? I.e. do we think it's common that people use this as a golang module? For them, I think, this change in function signature would be breaking stuff on upgrade. WDYT?

Reply:
You are right. I can add a new function `NewFromInterface`, for instance, that accepts the dynamic.Interface. And turn the function `New` into an stub that creates the interface from Config, and call `NewFromInterface`.

Reply:
Sounds reasonable, let's do that. 👍 Thanks!

Reply:
changes applied and branch rebased